src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,142 @@
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<Props> = ({
+  todos,
+  filter,
+  title,
+  iconId,
+  setIsModalOpen,
+  setTodoCard,
+  setIconId,
+}) => {
+  let filtering;
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  switch (filter) {
+    case FilteringType.active:
+      filtering = todos.filter((todo) => !todo.completed);
+      break;
+    case FilteringType.completed:
+      filtering = todos.filter((todo) => todo.completed);
+      break;
+    default: filtering = todos;
+  }
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  const findMatches = filtering
+    .filter((todo) => todo.title.trim().toLowerCase()
+      .includes(title.trim().toLowerCase()));
+
+  const handleEyeButton = (todo: Todo) => {
+    setTodoCard(todo);
+    setIsModalOpen(true);
+    setIconId(todo.id);
+  };
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+  return (
+    <table className="table is-narrow is-fullwidth">
+      <thead>
+        <tr>
+          <th>#</th>
+          <th>
             <span className="icon">
-              <i className="far fa-eye" />
+              <i className="fas fa-check" />
             </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+          </th>
+          <th>Title</th>
+          <th> </th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {title.trim().length > 0 ? (
```

Comment:
This component looks too complex and has a lot of duplicated code, better to create preparedTodos variable inside App component and pass prepared (filtered) todos to TodoList